### PR TITLE
*: support external query URL for Thanos ruler

### PR DESCRIPTION
**Pull request by simonpasquier**

### CODE REVIEW

File: pkg/apis/monitoring/v1/thanos_types.go [hunk 1]
```diff
@@ -170,6 +170,9 @@ type ThanosRulerSpec struct {
 	// Note: Currently only the CAFile, CertFile, and KeyFile fields are supported.
 	// Maps to the '--grpc-server-tls-*' CLI args.
 	GRPCServerTLSConfig *TLSConfig `json:"grpcServerTlsConfig,omitempty"`
+	// The external Query URL the Thanos Ruler will set in the 'Source' field
+	// of all alerts.
```
[the remark this answers is not in the document]
Comment (simonpasquier) [reply pkg/apis/monitoring/v1/thanos_types.go:174]: done

File: pkg/apis/monitoring/v1/thanos_types.go [hunk 2]
```diff
@@ -170,6 +170,9 @@ type ThanosRulerSpec struct {
...
+	ExternalQueryURL string `json:"externalQueryUrl,omitempty"`
```
[the remark this answers is not in the document]
Comment (simonpasquier) [reply pkg/apis/monitoring/v1/thanos_types.go]: done

**Comment by pgier** [comment]
@simonpasquier looks like something is weird with the bindata generation, you might need to delete the local `example/prometheus-operator-crd/monitoring.coreos.com_prometheus.yaml` and regenerate.

**Comment by simonpasquier** [comment]
@pgier thanks! lets see if it helps...

**Pull request merged**